migration/v0.18.x-v0.19.x/opensearch-migration-run.sh

```diff
@@ -0,0 +1,289 @@
+    exit 1
+fi
+
+sc_merged=$(yq m -x -a overwrite "${common_defaults}" "${sc_defaults}" "${common_config}" "${sc_config}")
+
+ES_BUCKET=$(echo "${sc_merged}" | yq r - "objectStorage.buckets.elasticsearch")
+ES_PASSWORD=$(sops --config "${sops_config}" -d "${secrets}" --extract '["elasticsearch"]["adminPassword"]')
+ES_REPOSITORY=$(echo "${sc_merged}" | yq r - "elasticsearch.snapshotRepository")
+
+OS_BUCKET=$(echo "${sc_merged}" | yq r - "objectStorage.buckets.opensearch")
+OS_PASSWORD=$(sops --config "${sops_config}" -d "${secrets}" --extract '["opensearch"]["adminPassword"]')
+OS_REPOSITORY=$(echo "${sc_merged}" | yq r - "opensearch.snapshot.repository")
+OS_CLUSTERNAME=$(echo "${sc_merged}" | yq r - "opensearch.clusterName")
+
+if [[ "${OS_BUCKET}" == "${ES_BUCKET}" ]]; then
+    echo -n "ERROR: OpenSearch cannot use the same bucket as ODFE."
+    exit 1
+elif [[ "${OS_REPOSITORY}" == "${ES_REPOSITORY}" ]]; then
+    echo -n "ERROR: OpenSearch cannot use the same repository name as ODFE."
+    exit 1
+fi
+
+BASE_DOMAIN=$(echo "${sc_merged}" | yq r - "global.baseDomain")
+OSD_SUBDOMAIN=$(echo "${sc_merged}" | yq r - "opensearch.dashboards.subdomain")
+
+echo "--- Migration from ODFE to OpenSearch ---"
+echo
+
+echo "--- Delete Fluentd Elasticsearch to stop incoming logs."
+echo "ck8s ops helmfile wc -l app=fluentd delete"
+echo -n "- run? [y/N]: "
+read -r reply
+if [[ "${reply}" != "y" ]]; then
+    exit 1
+fi
+"${here}/../../bin/ck8s" ops helmfile wc -l app=fluentd delete
+echo "---"
+echo
+
+echo "--- Take final ODFE snapshot ---"
+echo "ck8s ops kubectl sc -n elastic-system exec opendistro-es-master-0 -c elasticsearch -- \\"
+echo "    curl -XPUT http://localhost:9200/_snapshot/${ES_REPOSITORY}/final?wait_for_completion=true \\"
+echo "        -u admin:<password>"
+echo -n "- run? [y/N]: "
+read -r reply
+if [[ "${reply}" != "y" ]]; then
+    exit 1
+fi
+"${here}/../../bin/ck8s" ops kubectl sc -n elastic-system exec opendistro-es-master-0 -c elasticsearch -- \
+    curl -XPUT "http://localhost:9200/_snapshot/${ES_REPOSITORY}/final?wait_for_completion=true" \
+        -u "admin:${ES_PASSWORD}"
+echo "---"
+echo
+
+echo "--- Delete ODFE. (Skip if you want to keep it for now, remove it now if you are going to reuse the subdomains.)"
+echo "ck8s ops helmfile sc -l group=opendistro delete"
+echo -n "- run? [y/s/N]: "
+read -r reply
+if [[ "${reply}" == "y" ]]; then
+    "${here}/../../bin/ck8s" ops helmfile sc -l group=opendistro delete
+elif [[ "${reply}" == "s" ]]; then
+    echo "skipping"
+else
+    exit 1
+fi
+echo "---"
+echo
+
+echo "--- Update Dex with OpenSearch client. (Skip if it is not used.)"
+echo "ck8s ops helmfile sc -l app=dex apply"
+echo -n "- run? [y/s/N]: "
+read -r reply
+if [[ "${reply}" == "y" ]]; then
+    "${here}/../../bin/ck8s" ops helmfile sc -l app=dex apply
+elif [[ "${reply}" == "s" ]]; then
+    echo "skipping"
+else
+    exit 1
+fi
+echo "---"
+echo
+
+echo "--- Deploy OpenSearch."
```

Comment:
I really like the prompts in the script. But I feel like the non destructive once should be fine to just execute without it.

Reply:
Done. 
Saw I had double steps for some things which should not be there so I fixed that as well.